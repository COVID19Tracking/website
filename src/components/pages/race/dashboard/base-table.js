import React from 'react'
import { StateTable, StateTableHeader, StateTableBody } from './table'

const BaseTable = ({
  data,
  type,
  noPositives,
  noDeaths,
  noTests,
  noHospitalizations,
  rows,
}) => {
  return (
    <StateTable>
      <StateTableHeader
        groupTitle={type}
        noPositives={noPositives}
        noDeaths={noDeaths}
        noTests={noTests}
        noHospitalizations={noHospitalizations}
      />
      <StateTableBody
        noPositives={noPositives}
        noDeaths={noDeaths}
        noTests={noTests}
        noHospitalizations={noHospitalizations}
        state={data.name}
        stateAbbr={data.state}
        type={type}
        rows={rows}
      />
    </StateTable>
  )
}

export default BaseTable
