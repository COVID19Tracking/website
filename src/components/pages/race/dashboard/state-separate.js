import React from 'react'
import classnames from 'classnames'
import HeaderSorter from './header-sorter'
import TableNotes from './table-notes'
import PercentageOverview from './percentage-overview'
import anhpiNotes from './anhpi-notes'
import cautionNotes from './caution-notes'
import NoData from './no-data'
import RaceTable from './race-table'
import EthnicityTable from './ethnicity-table'
import stateStyle from './state.module.scss'

const StateSeparate = ({ state }) => {
  const stateData = state
  let raceNotes = {
    otherDeath: stateData.otherDeathNotes,
    otherPos: stateData.otherPosNotes,
    whiteDeath: stateData.whiteDeathNotes,
    whitePos: stateData.whitePosNotes,
    twoDeath: stateData.twoDeathNotes,
    twoPos: stateData.twoPosNotes,
    aianDeath: stateData.aianDeathNotes,
    aianPos: stateData.aianPosNotes,
    nhpiDeath: stateData.nhpiDeathNotes,
    nhpiPos: stateData.nhpiPosNotes,
    asianDeath: stateData.asianDeathNotes,
    asianPos: stateData.asianPosNotes,
    latinXDeath: stateData.latinXDeathNotes,
    latinXPos: stateData.latinXPosNotes,
    blackDeath: stateData.blackDeathNotes,
    blackPos: stateData.blackPosNotes,
    blackSpecialCase: stateData.blackSpecialCaseNotes,
    asianSpecialCase: stateData.asianSpecialCaseNotes,
    aianSpecialCase: stateData.aianSpecialCaseNotes,
    nhpiSpecialCase: stateData.nhpiSpecialCaseNotes,
    twoSpecialCase: stateData.twoSpecialCaseNotes,
    whiteSpecialCase: stateData.whiteSpecialCaseNotes,
    otherSpecialCase: stateData.otherSpecialCaseNotes,
  }

  raceNotes = anhpiNotes(stateData, raceNotes)
  raceNotes = cautionNotes(stateData, raceNotes)

  const groupedRaceNotes = [...new Set(Object.values(raceNotes))]
    .filter(value => value)
    .reverse()

  let ethnicityNotes = {
    nonhispanicPos: stateData.nonhispanicPosNotes,
    nonhispanicDeath: stateData.nonhispanicDeathNotes,
    latinXPos: stateData.latinXPosNotes,
    latinXDeath: stateData.latinXDeathNotes,
    latinXSpecialCase: stateData.latinXSpecialCaseNotes,
    nonhispanicSpecialCase: stateData.nonhispanicSpecialCaseNotes,
  }

  ethnicityNotes = cautionNotes(stateData, ethnicityNotes)

  const groupedEthnicityNotes = [...new Set(Object.values(ethnicityNotes))]
    .filter(value => value)
    .reverse()

  if (!stateData.anyPosData && !stateData.anyDeathData) {
    return <NoData stateName={stateData.name} />
  }

  const disparityExists =
    Object.keys(stateData).filter(
      field => field.search('DispFlag') > -1 && stateData[field],
    ).length > 0

  return (
    <div>
      <div className={stateStyle.stateOverview}>
        {/* this is the race data section */}
        <PercentageOverview
          stateName={state.name}
          dataType="race"
          casePercent={state.knownRacePos}
          deathPercent={state.knownRaceDeath}
          className={stateStyle.totals}
        />
        <div className={stateStyle.note}>
          <HeaderSorter stateName={state.name} stateReports="race" />
        </div>
      </div>
      <h3 className={stateStyle.tableTitle}>Cases and deaths by race</h3>
      <RaceTable
        data={stateData}
        type="Race"
        notes={raceNotes}
        groupedNotes={groupedRaceNotes}
        noPositives={!stateData.posRaceData}
        noDeaths={!stateData.deathRaceData}
        isInEthnicityState
      />
      <TableNotes
        state={stateData.state}
        stateName={stateData.name}
        type="race"
        groupedNotes={groupedRaceNotes}
        disparityExists={disparityExists}
      />
      <>
        <div
          className={classnames(
            stateStyle.stateOverview,
            stateStyle.ethnicityOverview,
          )}
        >
          {/* this is the ethnicity data section */}
          <PercentageOverview
            stateName={state.name}
            dataType="ethnicity"
            casePercent={state.knownEthPos}
            deathPercent={state.knownEthDeath}
            className={stateStyle.totals}
          />
          <div className={stateStyle.note}>
            <HeaderSorter stateName={state.name} stateReports="ethnicity" />
          </div>
        </div>
        {(stateData.posEthData || stateData.deathEthData) && (
          <>
            <h3 className={stateStyle.tableTitle}>
              Cases and deaths by ethnicity
            </h3>
            <EthnicityTable
              data={stateData}
              type="Ethnicity"
              notes={ethnicityNotes}
              groupedNotes={groupedEthnicityNotes}
              noPositives={!stateData.posEthData}
              noDeaths={!stateData.deathEthData}
            />
            <TableNotes
              state={stateData.state}
              stateName={stateData.name}
              type="ethnicity"
              groupedNotes={groupedEthnicityNotes}
              disparityExists={disparityExists}
            />
          </>
        )}
      </>
    </div>
  )
}

export default StateSeparate
