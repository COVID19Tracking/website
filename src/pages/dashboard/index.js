import React from 'react'
import Layout from '../../components/layout'
import Map from './_Map'

const DashboardPage = () => (
  <Layout title="Visual Dashboard">
    <p>Test</p>
    <Map />
  </Layout>
)

export default DashboardPage
