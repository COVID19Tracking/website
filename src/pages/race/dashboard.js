import React from 'react'
import { graphql } from 'gatsby'
import Layout from '~components/layout'

import StateNav from '~components/pages/race/dashboard/state-nav'
import LongContent from '~components/common/long-content'

import States from '~components/pages/race/dashboard/states'
import UsOverview from '~components/pages/race/dashboard/us-overview'
import Preamble from '~components/pages/race/dashboard/preamble'

const RaceDashboardPage = ({ data }) => {
  const stateList = []
  data.allCovidRaceDataSeparate.nodes.forEach(state => {
    stateList.push(state)
  })
  data.allCovidRaceDataCombined.nodes.forEach(state => {
    stateList.push(state)
  })
  return (
    <Layout
      title="Racial Data Dashboard"
      returnLinks={[{ link: '/race', title: 'Racial Data Tracker' }]}
      path="/race/dashboard"
      socialCard={data.contentfulSocialCard}
    >
      <LongContent>
        <Preamble
          raceHeroSnippetHtml={
            data.raceHeroSnippet.childContentfulSnippetContentTextNode
              .childMarkdownRemark.html
          }
        />
        <UsOverview
          statesCasesCount={data.covidRaceDataHomepage.statesReportingCases}
          statesDeathsCount={data.covidRaceDataHomepage.statesReportingDeaths}
        />
      </LongContent>
      <StateNav
        title="Race and ethnicity data by state"
        stateList={stateList.sort((a, b) => (a.name < b.name ? -1 : 1))}
      />
      <States />
    </Layout>
  )
}

export default RaceDashboardPage

export const query = graphql`
  query {
    raceHeroSnippet: contentfulSnippet(slug: { eq: "race-hero-lede" }) {
      childContentfulSnippetContentTextNode {
        childMarkdownRemark {
          html
        }
      }
    }
    contentfulSocialCard(slug: { eq: "racial-data-tracker" }) {
      description {
        description
      }
      image {
        resize(width: 1200) {
          src
        }
      }
    }
    allCovidRaceDataSeparate {
      nodes {
        name
        state
      }
    }
    allCovidRaceDataCombined {
      nodes {
        name
        state
      }
    }
    covidRaceDataHomepage {
      statesReportingCases
      statesReportingDeaths
    }
    allCovidStateInfo {
      nodes {
        state
        name
      }
    }
  }
`
