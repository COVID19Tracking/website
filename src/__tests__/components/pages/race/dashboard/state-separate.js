import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import { StateSeparate } from '~components/pages/race/dashboard/state-separate'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    disparityNote: {
      contentful_id: 'aNeWRaNdoMSTrinG',
      childContentfulSnippetContentTextNode: {
        childMarkdownRemark: {
          html: '<h2>Content</h2><p>More disparity snippet content</p>',
        },
      },
    },
    comparibleNote: {
      contentful_id: 'RaNdoMSTrinG',
      childContentfulSnippetContentTextNode: {
        childMarkdownRemark: {
          html: '<h2>Content</h2><p>More comparible snippet content</p>',
        },
      },
    },
  }))
})

describe('Components : Pages : Race : Dashboard : State Separate', () => {
  it('renders correctly', () => {
    const state = {
      aianDeathNotes: 'Race notes: aianDeath',
      aianPosNotes: 'Race notes: aianPos',
      aianSpecialCaseNotes: 'Race notes: aianSpecialCase',
      anyDeathData: true,
      anyPosData: true,
      asianDeathNotes: 'Race notes: asianDeath',
      asianPosNotes: 'Race notes: asianPos',
      asianSpecialCaseNotes: 'Race notes: asianSpecialCase',
      blackDeathNotes: 'Race notes: blackDeath',
      blackPosNotes: 'Race notes: blackPos',
      blackSpecialCaseNotes: 'Race notes: blackSpecialCase',
      deathEthData: false,
      deathRaceData: true,
      latinXDeathNotes: 'Race notes: latinXDeath',
      latinXPosNotes: 'Race notes: latinXPos',
      latinXSpecialCaseNotes: 'Race notes: latinXSpecialCase',
      name: 'Alabama',
      nhpiDeathNotes: 'Race notes: nhpiDeath',
      nhpiPctPop: 0.36,
      nhpiPctPos: 0.263,
      nhpiPosCaution: true,
      nhpiPosDispFlag: false,
      nhpiPosNotes: 'Race notes: nhpiPos',
      nhpiSpecialCaseNotes: 'Race notes: nhpiSpecialCase',
      nonhispanicDeathNotes: 'Race notes: nonhispanicDeath',
      nonhispanicPosNotes: 'Race notes: nonhispanicPos',
      nonhispanicSpecialCaseNotes: 'Race notes: nonhispanicSpecialCase',
      otherDeathCaution: true,
      otherDeathDispFlag: false,
      otherDeathNotes: 'Race notes: otherDeath',
      otherPctDeath: 0.123,
      otherPctPop: 0.31,
      otherPctPos: 0.857,
      otherPosCaution: true,
      otherPosDispFlag: false,
      otherPosNotes: 'Race notes: otherPos',
      otherSpecialCaseNotes: 'Race notes: otherSpecialCase',
      posEthData: true,
      posRaceData: false,
      state: 'AL',
      twoDeathCaution: true,
      twoDeathDispFlag: false,
      twoDeathNotes: 'Race notes: twoDeath',
      twoPctDeath: 'twoPctDeath',
      twoPctPop: 0.989,
      twoPctPos: 0.379,
      twoPosCaution: true,
      twoPosDispFlag: false,
      twoPosNotes: 'Race notes: twoPos',
      twoSpecialCaseNotes: 'Race notes: twoSpecialCase',
      whiteDeathCaution: true,
      whiteDeathDispFlag: false,
      whiteDeathNotes: 'Race notes: whiteDeath',
      whitePctDeath: 'whitePctDeath',
      whitePctPop: 0.033,
      whitePctPos: 0.787,
      whitePosCaution: true,
      whitePosDispFlag: false,
      whitePosNotes: 'Race notes: whitePos',
      whiteSpecialCaseNotes: 'Race notes: whiteSpecialCase',
    }
    const tree = renderer.create(<StateSeparate state={state} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly', () => {
    const state = {
      anyPosData: false,
      anyDeathData: false,
    }
    const noDataTree = renderer.create(<StateSeparate state={state} />).toJSON()
    expect(noDataTree).toMatchSnapshot()
  })
})
