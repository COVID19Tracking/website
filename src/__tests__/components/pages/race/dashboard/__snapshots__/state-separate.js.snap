// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Race : Dashboard : State Separate renders correctly 1`] = `
<div>
  <div
    className="stateOverview"
  >
    <div
      className="note"
    >
      <div
        className="container"
      >
        <p>
          The following tables reflect only those cases and deaths where
           
          race
           is known and reported by 
          Alabama
          . If this stateâ€™s reporting percentages are low, interpret with caution.
        </p>
      </div>
    </div>
  </div>
  <h3
    className="tableTitle"
  >
    Cases and deaths by race
  </h3>
  <div>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            className="group"
            scope="col"
          >
            <span
              role="text"
            >
              Race
            </span>
          </th>
          <th
            className="borderLeft percent"
            scope="col"
          >
            <span
              role="text"
            >
              Percentage of population
            </span>
          </th>
          <th
            className="percent missingData"
            scope="col"
          >
            <span
              role="text"
            >
              <img
                alt="Alert icon"
                src="test-file-stub"
              />
              Percentage of cases
              <div
                className="symbolSpacer"
              />
            </span>
          </th>
          <th
            className="percent"
            scope="col"
          >
            <span
              role="text"
            >
              Percentage of deaths
              <div
                className="symbolSpacer"
              />
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Black or African American alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            0
            %
          </td>
          <td
            className="missingData"
            rowSpan={7}
          >
            <span>
              Alabama
               does not report 
              race
               data for 
              cases
              .
               
              <a
                href="/race/get-better-data"
              >
                Help us get better data.
              </a>
            </span>
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    Race notes: blackDeath
                  </span>
                  <span>
                    9
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Asian alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            0
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    Race notes: asianDeath
                  </span>
                  <span>
                    13
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Native Hawaiian and Pacific Islander alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            36
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    Race notes: nhpiDeath
                  </span>
                  <span>
                    15
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              American Indian or Alaska Native alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            0
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    Race notes: aianDeath
                  </span>
                  <span>
                    17
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Two or more races
            </span>
          </th>
          <td
            className="borderLeft"
          >
            99
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Race notes: twoDeath
                  </span>
                  <span>
                    19
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              White alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            3
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Race notes: whiteDeath
                  </span>
                  <span>
                    21
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Some other race alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            31
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                12
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Race notes: otherDeath
                  </span>
                  <span>
                    23
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <h4
    className="header"
    id="notes-al"
  >
    Notes
     
    <span
      className="a11y-only"
    >
      on 
      race
       for 
      Alabama
    </span>
  </h4>
  <ul
    className="list"
  >
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          1
        </span>
      </span>
       
      Race notes: otherSpecialCase
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          2
        </span>
      </span>
       
      Race notes: whiteSpecialCase
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          3
        </span>
      </span>
       
      Race notes: twoSpecialCase
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          4
        </span>
      </span>
       
      Race notes: nhpiSpecialCase
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          5
        </span>
      </span>
       
      Race notes: aianSpecialCase
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          6
        </span>
      </span>
       
      Race notes: asianSpecialCase
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          7
        </span>
      </span>
       
      Race notes: blackSpecialCase
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          8
        </span>
      </span>
       
      Race notes: blackPos
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          9
        </span>
      </span>
       
      Race notes: blackDeath
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          10
        </span>
      </span>
       
      Race notes: latinXPos
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          11
        </span>
      </span>
       
      Race notes: latinXDeath
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          12
        </span>
      </span>
       
      Race notes: asianPos
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          13
        </span>
      </span>
       
      Race notes: asianDeath
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          14
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Race notes: nhpiPos
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          15
        </span>
      </span>
       
      Race notes: nhpiDeath
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          16
        </span>
      </span>
       
      Race notes: aianPos
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          17
        </span>
      </span>
       
      Race notes: aianDeath
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          18
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Race notes: twoPos
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          19
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Race notes: twoDeath
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          20
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Race notes: whitePos
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          21
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Race notes: whiteDeath
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          22
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Race notes: otherPos
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          23
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Race notes: otherDeath
    </li>
  </ul>
  <div
    className="stateOverview ethnicityOverview"
  >
    <div
      className="note"
    >
      <div
        className="container"
      >
        <p>
          The following tables reflect only those cases and deaths where
           
          ethnicity
           is known and reported by 
          Alabama
          . If this stateâ€™s reporting percentages are low, interpret with caution.
        </p>
      </div>
    </div>
  </div>
  <h3
    className="tableTitle"
  >
    Cases and deaths by ethnicity
  </h3>
  <div>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            className="group"
            scope="col"
          >
            <span
              role="text"
            >
              Ethnicity
            </span>
          </th>
          <th
            className="borderLeft percent"
            scope="col"
          >
            <span
              role="text"
            >
              Percentage of population
            </span>
          </th>
          <th
            className="percent"
            scope="col"
          >
            <span
              role="text"
            >
              Percentage of cases
              <div
                className="symbolSpacer"
              />
            </span>
          </th>
          <th
            className="percent missingData"
            scope="col"
          >
            <span
              role="text"
            >
              <img
                alt="Alert icon"
                src="test-file-stub"
              />
              Percentage of deaths
              <div
                className="symbolSpacer"
              />
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Hispanic or Latino
            </span>
          </th>
          <td
            className="borderLeft"
          >
            0
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    Race notes: latinXPos
                  </span>
                  <span>
                    4
                  </span>
                </a>
              </span>
            </div>
          </td>
          <td
            className="missingData"
            rowSpan={2}
          >
            <span>
              Alabama
               does not report 
              ethnicity
               data for 
              deaths
              .
               
              <a
                href="/race/get-better-data"
              >
                Help us get better data.
              </a>
            </span>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Not Hispanic or Latino
            </span>
          </th>
          <td
            className="borderLeft"
          >
            0
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    Race notes: nonhispanicPos
                  </span>
                  <span>
                    6
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <h4
    className="header"
    id="notes-al"
  >
    Notes
     
    <span
      className="a11y-only"
    >
      on 
      ethnicity
       for 
      Alabama
    </span>
  </h4>
  <ul
    className="list"
  >
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          1
        </span>
      </span>
       
      Race notes: nonhispanicSpecialCase
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          2
        </span>
      </span>
       
      Race notes: latinXSpecialCase
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          3
        </span>
      </span>
       
      Race notes: latinXDeath
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          4
        </span>
      </span>
       
      Race notes: latinXPos
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          5
        </span>
      </span>
       
      Race notes: nonhispanicDeath
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          6
        </span>
      </span>
       
      Race notes: nonhispanicPos
    </li>
  </ul>
</div>
`;

exports[`Components : Pages : Race : Dashboard : State Separate renders correctly 2`] = `
<div
  className="container"
>
  <table
    aria-hidden={true}
    className="table"
    role="presentation"
  >
    <tbody>
      <tr>
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
      </tr>
    </tbody>
  </table>
  <div
    className="alert"
  >
    <div
      className="title"
    >
      <img
        alt="Alert icon"
        src="test-file-stub"
      />
      <span>
        No Data
      </span>
    </div>
    <div
      className="content"
    >
      <p>
         does not share data about race or ethnicity.
         
        <a
          href="/race/get-better-data"
        >
          Help us advocate for better data
        </a>
        .
      </p>
    </div>
  </div>
</div>
`;
